README.md

```diff
@@ -1,2 +1,43 @@
-# opentoken-lib-js
-A library to interact with the opentoken API.
+OpenToken Library
+=================
+
+A library to interact with the [OpenToken API](https://github.com/opentoken-io/opentoken).
+
+
+Usage
+-----
+
+The library exposes two functions: `createSignedTokenRequests` and `createTokenRequest`. Let's go through an example of a common use case for token requests.
+
+    // Instantiate the library.
+    var lib = require("opentoken-lib")("api.opentoken.io");
+
+    // Create signedTokenRequests.
+    var signedTokenRequests = lib.createSignedTokenRequests("accountId", "openTokenCode", "secretCode");
+
+    // Create tokenRequest for downloading public content. OpenToken code and secret are not required.
+    var tokenRequest = lib.createTokenRequest("accountId");
+
+    // Upload content to public endpoint. Takes a stream or file name.
+    signedTokenRequests.public.uploadFromFile("keep-me-safe.txt").then((token) => {
+        // Upload the public token to a private endpoint.
+        return signedTokenRequests.private.upload(token);
+    }).then((token) => {
+        // Grab public token from private endpoint.
+        return signedTokenRequests.private.download(token);
```

Comment:
Here is the breakdown on that, `private` and `public` are instances of the same class. The only real difference is the property `isPublic` which is used in the query string for uploads. The downloads are the same. We saw no purpose in breaking this up further design wise because it would become more confusing.